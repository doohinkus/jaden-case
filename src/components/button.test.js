import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import Button from './button';
Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
  it('should pass text to the text prop and a function to the action prop', () => {
    const action = jest.fn();
    const text = "text";
    const output = shallow(
      <Button action={action} text={text} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
})
