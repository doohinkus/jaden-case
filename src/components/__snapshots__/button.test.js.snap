// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should pass text to the text prop and a function to the action prop 1`] = `
<button
  className="btn btn-primary btn-block"
  onClick={[Function]}
  target="_blank"
>
  text
</button>
`;

exports[`Button should render button 1`] = `
<button
  className="btn btn-primary btn-block"
  onClick={[Function]}
  target="_blank"
>
  text
</button>
`;
